kafka-avro-adapter/src/cs_producer.cpp

```diff
@@ -18,140 +18,177 @@
 #include "common.h"
 
 void CSProducer::processAvroType(BulkInsert& insert, avro_value_t* value)
-
 {
-    auto table = m_driver.getSystemCatalog().getTable(m_options.database, m_options.table);
-
-    assert(avro_value_get_type(value) == AVRO_RECORD);
-    size_t n_fields = 0;
-    avro_value_get_size(value, &n_fields);
+    mcsapi::ColumnStoreSystemCatalogTable table = m_driver.getSystemCatalog().getTable(m_options.database, m_options.table);
```

Comment:
Don't change the variable type unless necessary (auto is pretty convenient when you just pass the variable as a parameter).
```suggestion
    auto table = m_driver.getSystemCatalog().getTable(m_options.database, m_options.table);
```